src/client.c

```diff
@@ -432,6 +432,15 @@ bool set_pool(PgSocket *client, const char *dbname, const char *username, const
 		strlcpy(client->db->name, dbname, sizeof(client->db->name));
 	}
 
+	if (client->db->max_db_client_connections > 0) {
```

Comment:
This is currently ignoring the global `max_db_client_connections` setting.

```suggestion
	if (database_max_client_connections(client->db) > 0) {
```